### PR TITLE
feat: Syntax errors

### PR DESCRIPTION
**Pull request by Suyashtnt**
Adds syntax errors.
This extension is now verified better than the official one:tm:

### CODE REVIEW

**Comment by Suyashtnt** [comment]
Showcase: 

[Le showcase](https://www.youtube.com/watch?v=bKPYtz1rgMk)

**Comment by krista-chan** [comment]
looking smexy :3

File: src/main.rs [hunk 1]
```diff
@@ -53,3 +85,100 @@ async fn main() {
...
+                file: _,
+            } => {
+                client
+                    .publish_diagnostics(
+                        text_location,
+                        vec![Diagnostic {
+                            code: None,
+                            code_description: None,
+                            data: None,
+                            message: format!("ERROR: expected {}\nFOUND: {}", expected, found),
```
Comment (Spu7Nix) [inline src/main.rs:132]: Maybe a comma here: `"ERROR: expected {},\nFOUND: {}"`

**Review by Spu7Nix — APPROVED** [review]
Epic error moment

File: src/main.rs (hunk 1 above)
Comment (Suyashtnt) [reply src/main.rs]: done

Comment (krista-chan) [reply src/main.rs]: done'd

**Review by krista-chan — APPROVED** [review]
All code looks good to me

**Pull request merged**